### PR TITLE
added declaration file

### PR DESCRIPTION
**Pull request by koocieyu**
this solves #81

### CODE REVIEW

**Comment by koocieyu** [comment]
this PR solves #81

**Comment by koocieyu** [comment]
saw some methods that need to have their parameters marked as optional, will fix tomorrow

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+    * import FileHound from 'filehound';
+    *
+    * const filehound = FileHound.create();
+    * const files = filehound.findSync();
+    * console.log(files);
+    */
+    public findSync(): string[];
+    
+    /**
+    * @memberof _ (lodash)
```
Comment (nspragg) [inline index.d.ts]: `each` is a method on the promise returned but `find`. It's not actually a method on `filehound`

Comment (koocieyu) [reply index.d.ts]: thanks a lot for the clarification

**Comment by nspragg** [comment]
@koocieyu - thanks for your contribution.

**Pull request merged**